core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,179 @@
+    Show.show[NonEmptyVector[A]](_.show)
+
+  implicit def catsDataSemigroupKForNonEmptyVector: SemigroupK[NonEmptyVector] =
+    new SemigroupK[NonEmptyVector] {
+      def combineK[A](a: NonEmptyVector[A], b: NonEmptyVector[A]): NonEmptyVector[A] =
+        a combine b
+    }
+
+  implicit def catsDataSemigroupForNonEmptyVector[A]: Semigroup[NonEmptyVector[A]] =
+    catsDataSemigroupKForNonEmptyVector.algebra
+
+  implicit def catsDataReducibleForNonEmptyVector: Reducible[NonEmptyVector] =
+    new NonEmptyReducible[NonEmptyVector, Vector] {
+      override def split[A](fa: NonEmptyVector[A]): (A, Vector[A]) = (fa.head, fa.tail)
+
+      override def size[A](fa: NonEmptyVector[A]): Long = 1 + fa.tail.size.toLong
+    }
+
+  implicit def catsDataMonadForNonEmptyVector: Monad[NonEmptyVector] =
+    new Monad[NonEmptyVector] {
+      override def map[A, B](fa: NonEmptyVector[A])(f: A => B): NonEmptyVector[B] =
+        fa map f
+
+      def pure[A](x: A): NonEmptyVector[A] =
+        NonEmptyVector(x, Vector.empty)
+
+      def flatMap[A, B](fa: NonEmptyVector[A])(f: A => NonEmptyVector[B]): NonEmptyVector[B] =
+        fa flatMap f
+    }
+}
+
+trait NonEmptyVectorLowPriority0 {
+
+  implicit def catsDataComonadForNonEmptyVector: Comonad[NonEmptyVector] = new Comonad[NonEmptyVector] {
+
+    def coflatMap[A, B](fa: NonEmptyVector[A])(f: NonEmptyVector[A] => B): NonEmptyVector[B] = {
+        @tailrec def consume(as: Vector[A], buf: VectorBuilder[B]): Vector[B] =
+          as match {
+            case a +: as => consume(as, buf += f(NonEmptyVector(a, as)))
+            case _ => buf.result()
+          }
+        NonEmptyVector(f(fa), consume(fa.tail, new VectorBuilder[B]))
+      }
+
+
+    def extract[A](fa: NonEmptyVector[A]): A = fa.head
+
+    def map[A, B](fa: NonEmptyVector[A])(f: A => B): NonEmptyVector[B] = fa map f
+  }
+}
+
+trait NonEmptyVectorLowPriority1 extends NonEmptyVectorLowPriority0 {
+  implicit def catsDataFunctorForNonEmptyVector: Functor[NonEmptyVector] =
+    new Functor[NonEmptyVector] {
+      def map[A, B](fa: NonEmptyVector[A])(f: A => B): NonEmptyVector[B] =
+        fa map f
+    }
+
+}
+
+trait NonEmptyVectorLowPriority2 extends NonEmptyVectorLowPriority1 {
+  implicit def catsDataTraverseForNonEmptyVector: Traverse[NonEmptyVector] =
+    new Traverse[NonEmptyVector] {
+      def traverse[G[_], A, B](fa: NonEmptyVector[A])(f: (A) => G[B])(implicit G: Applicative[G]): G[NonEmptyVector[B]] =
+      G.map2Eval(f(fa.head), Always(Traverse[Vector].traverse(fa.tail)(f)))(NonEmptyVector(_, _)).value
+
+
+      def foldLeft[A, B](fa: NonEmptyVector[A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b)(f)
+
+      def foldRight[A, B](fa: NonEmptyVector[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+        fa.foldRight(lb)(f)
+    }
+}
+
+object NonEmptyVector extends NonEmptyVectorInstances {
+
+  def apply[A](head: A, tail: Vector[A]): NonEmptyVector[A] = NonEmptyVector(head +: tail)
```

Comment:
I would prefer to support this as:

```scala
def apply[A](body: Vector[A], last: A): NonEmptyVector[A] = NonEmptyVector(body :+ last)
```

In general I don't like treating non-empty vector as a head/tail pair because it's so inefficient -- prepending to a vector is O(n) and reallocates the entire vector.

Honestly I'd probably prefer an interface like:

```scala
def apply[A](xs: Vector[A]): Option[NonEmptyVector[A]] = ...
```